### PR TITLE
Fix/codesearch optimizations and tests

### PR DESCRIPTION
**Pull request by sanzog03**
## Issue
1. `@lru_cache` decorator on async functions causing `RuntimeError: cannot reuse already awaited coroutine` when cached coroutines are awaited multiple times
2. Missing null safety check in [calculate_reliability_score](cci:1://file:///Users/thapa24-mbp/Devs/akd/akd-ext/akd_ext/tools/code_search/utils.py:62:0-136:32) causing potential `AttributeError` 
3. Insufficient test coverage for code search utilities

## Resolution
1. **Fixed async caching mechanism**: Replaced `@lru_cache` with `@async_lru_cache` from the `akd.utils.async_lru_cache` package for the [fetch_github_metadata](cci:1://file:///Users/thapa24-mbp/Devs/akd/akd-ext/akd_ext/tools/code_search/utils.py:34:0-59:30) function. The `@lru_cache` caches the coroutine object itself rather than the result, while `@alru_cache` properly caches the awaited results of async functions, allowing concurrent requests with identical arguments.
2. **Improved reliability score calculation**: Added null safety check to handle `None` input in [calculate_reliability_score](cci:1://file:///Users/thapa24-mbp/Devs/akd/akd-ext/akd_ext/tools/code_search/utils.py:62:0-136:32):
   ```python
   if not repository_metadata or repository_metadata.is_null_metadata:
       return None
       
3. **Added comprehensive test coverage**:
    - Created tests/tools/code_search/test_utils.py with tests for successful metadata fetching, failure handling for non-existent repositories, reliability score calculations, and null metadata edge cases
    - Enhanced tests/tools/code_search/test_repository_search.py with parametrized tests for page size variations (3, 8, 10, 11, 55, 100) to verify results are capped at maximum 10 regardless of page_size setting

### CODE REVIEW

**Comment by NISH1001** [comment]
@sanzog03 we have https://github.com/NASA-IMPACT/accelerated-discovery/blob/develop/akd/utils.py#L19
this already in akd core async lru cache. can you try that since it's arlady in akd core so.

File: akd_ext/tools/code_search/utils.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def is_null_metadata(self) -> bool:
         )
 
 
-@lru_cache(maxsize=128)
+@alru_cache(maxsize=128)
```
Comment (NISH1001) [inline akd_ext/tools/code_search/utils.py]: we also have `akd.utils.async_lru_cache` which could be self-contained? no external deps

Comment (sanzog03) [reply akd_ext/tools/code_search/utils.py]: changed!

**Comment by sanzog03** [comment]
> @sanzog03 we have https://github.com/NASA-IMPACT/accelerated-discovery/blob/develop/akd/utils.py#L19 this already in akd core async lru cache. can you try that since it's arlady in akd core so.

changed!

File: uv.lock
Comment (NISH1001) [inline uv.lock:1]: Is this the updated uv lock? What changes occur here? did you add new pcakagges?

Comment (NISH1001) [reply uv.lock:1]: I just want to make sure we're not pushing unnecessary local package changes that are not required for akd-ext functioning

Comment (sanzog03) [reply uv.lock:1]: yes. updated the uv lock after removing the unwanted package.

**Review by NISH1001 — APPROVED** [review]
